src/actions/write/submitAuctionBid.ts

```diff
@@ -162,20 +162,19 @@ export const submitAuctionBid = (
     };
 
     /**
-     * Give the unsettled value to the protocol (it should already have the floor price from the initiated auction)
+     * Give the total value to the protocol
      * Deduct the unsettled final bid value from the caller
-     * Return floor price from the protocol balance to the initiator, if necessary
+     * Return floor price from the auction's vaulted balance to the initiator, if necessary
      */
-    updatedBalances[SmartWeave.contract.id] += finalBidForCaller.valueOf();
+    updatedBalances[SmartWeave.contract.id] +=
+      currentRequiredMinimumBid.valueOf();
     updatedBalances[caller] -= finalBidForCaller.valueOf();
 
     if (caller !== existingAuction.initiator) {
       // pull in the initiators existing balance and update it
       updatedBalances[existingAuction.initiator] =
-        (state.balances[existingAuction.initiator] || 0) +
+        (updatedBalances[existingAuction.initiator] || 0) +
         existingAuction.floorPrice;
```

Comment:
this is a difficult use case. we need to use the updatedBalances of the initiator if we have it, otherwise we need to use the `state.balances` balance.

Reply:
```
    if (caller !== existingAuction.initiator) {
      const initiatorsNewBalanceBalance =
        (updatedBalances[existingAuction.initiator] ||
          state.balances[existingAuction.initiator] ||
          0) + existingAuction.floorPrice;
      // pull in the initiators existing balance and update it
      updatedBalances[existingAuction.initiator] = initiatorsNewBalanceBalance;
    }
```

Reply:
[b5575d3](https://github.com/ar-io/testnet-contract/pull/124/commits/b5575d337b405273a3669cf9360a0023734e826a)